kani-driver/src/call_cargo.rs

```diff
@@ -123,15 +129,42 @@ fn glob(path: &Path) -> Result<Vec<PathBuf>> {
 ///   - I.e.: Do whatever cargo does when there's no default_members.
 ///   - This is because `default_members` is not available in cargo metadata.
 ///     See <https://github.com/rust-lang/cargo/issues/8033>.
-fn project_members(args: &KaniArgs, metadata: &Metadata) -> Vec<String> {
+fn project_members<'a, 'b>(args: &'a KaniArgs, metadata: &'b Metadata) -> Vec<&'b Package> {
     if !args.package.is_empty() {
-        args.package.clone()
+        args.package
+            .iter()
+            .map(|pkg_name| {
+                metadata
+                    .packages
+                    .iter()
+                    .find(|pkg| pkg.name == *pkg_name)
+                    .expect(&format!("Cannot find package '{}'", pkg_name))
+            })
+            .collect()
     } else {
         match (args.workspace, metadata.root_package()) {
-            (true, _) | (_, None) => {
-                metadata.workspace_members.iter().map(|id| metadata[id].name.clone()).collect()
-            }
-            (_, Some(root_pkg)) => vec![root_pkg.name.clone()],
+            (true, _) | (_, None) => metadata.workspace_packages(),
+            (_, Some(root_pkg)) => vec![root_pkg],
         }
     }
 }
+
+/// Extract the targets inside a package.
+/// If `--tests` is given, the list of targets will include any integration tests.
+fn package_targets(args: &KaniArgs, package: &Package) -> Vec<Vec<String>> {
```

Comment:
I thought about creating a type, but it would end up being a bit messy, since you end up with 2 args for `--tests` and `--bin` but only one for `--lib`. This would create more branches later when translating it to arguments. So I ended up keeping it simple and just using `Vec<Vec<_>>`.